src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -31,22 +34,42 @@ namespace Microsoft.PowerFx.Core.Functions
     /// This includings the binding (and hence IR for evaluation) - 
     /// This is conceptually immutable after initialization - if the body or signature changes, you need to create a new instance.
     /// </summary>
-    internal class UserDefinedFunction : TexlFunction
+    internal class UserDefinedFunction : TexlFunction, IExternalPageableSymbol, IExternalDelegatableSymbol
     {
         private readonly bool _isImperative;
         private readonly IEnumerable<UDFArg> _args;
         private TexlBinding _binding;
 
-        public override bool IsAsync => _binding?.IsAsync(UdfBody) ?? false;
+        public override bool IsAsync => _binding.IsAsync(UdfBody);
+
+        public bool IsPageable => _binding.IsPageable(_binding.Top);
+
+        public bool IsDelegatable => _binding.IsDelegatable(_binding.Top);
+
+        public override bool IsServerDelegatable(CallNode callNode, TexlBinding binding)
+        {
+            Contracts.AssertValue(callNode);
+            Contracts.AssertValue(binding);
+            Contracts.Assert(binding.GetInfo(callNode).Function is not UserDefinedFunction udf || udf.Binding != null);

```

Comment:
>  Contracts.Assert(binding.GetInfo(callNode).Function is not UserDefinedFunction udf || udf.Binding != null); [](http://example.com/codeflow?start=10&length=109)

This assert needs to be

  Contracts.Assert(binding.GetInfo(callNode).Function is UserDefinedFunction udf && udf.Binding != null);